### ISSUE TITLE
JSpecifyExperimental: false positive for nullable array already checked to be non-null

### ISSUE THREAD

**Issue by unv-unv** [uber/NullAway#1821]
JSpecifyExperimental mode on.

```java
@NullMarked
class Test {
    public static <T> T @Nullable [] copyOf(T @Nullable [] original) {
        return original == null 
            ? null 
            : Arrays.copyOf(
                original /* BUG: type variable T is constrained to be @Nullable, but its upper bound requires it to be @NonNull */, 
                original.length
            );
    }
}
```
Shows compilation error:
```
[ERROR]Test.java[7,17]: [NullAway] inference failure: type variable T is constrained to be @Nullable, but its upper bound requires it to be @NonNull
```
NullAway 0.14.1, Error Prone 2.50.0.

**Comment by msridhar** [comment]
`T @Nullable []` means the pointer to the array itself can be null.  Did you mean that the array elements can be null?  If you, so want `@Nullable T []`.  (See https://jspecify.dev/docs/user-guide/#type-use-annotation-syntax).

In your example, if you mean that `original` can be `null`, `Arrays.copyOf` is [documented](https://github.com/jspecify/jdk/blob/dc1f92c844ec566f7ba78b40c5d2e9282b70fa88/src/java.base/share/classes/java/util/Arrays.java#L3483) to throw an NPE if the array is `null`.

**Comment by msridhar** [comment]
Again, though, the error message is bad here, sorry about that.  I'll look into it.
